PetaPoco/Database.cs

```diff
@@ -673,6 +689,19 @@ private void SetParameterProperties(IDbDataParameter p, object value, PropertyIn
                     // Thanks @DataChomp for pointing out the SQL Server indexing performance hit of using wrong string type on varchar
                     p.DbType = DbType.AnsiString;
                 }
+                else if (t == typeof(DateTime2))
+                {
+                    var asValue = (value as DateTime2).Value;
+                    if (asValue == null)
```

Comment:
This `if..then` block can be a single line:

```c#
p.Value = asValue ?? DBNull.Value;
```

Reply:
?? operator dont work with DateTime

Reply:
Good point. Then `asValue` will never be null, so you don't need a null check. I think you can just do `p.Value = (value as DateTime2).Value`.

Reply:
the source can be nullable DateTime object, null check has to be done

Reply:
Oh, are you worried about the `value` argument being null? In that case, your line 694 would throw a null reference exception (trying to check `Value` property on a null object).

This should work:

```c#
p.Value = (value as DateTime2)?.Value ?? DBNull.Value;

// or

var asValue = (value as DateTime2)?.Value;
p.Value = asValue ?? DBNull.Value;
```

Reply:
Compiler Error CS0019 on both

Reply:
Whoops, need to cast `DBNull.Value` to `object`.

```c#
p.Value = asValue ?? (object)DBNull.Value;
```

Reply:
fixed in new commit